specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2021-04-01/ThreatIntelligence.json

```diff
@@ -0,0 +1,1173 @@
+    "version": "2021-04-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{operationalInsightsResourceProvider}/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/threatIntelligence/main/createIndicator": {
+      "post": {
+        "x-ms-examples": {
+          "Create a new Threat Intelligence": {
+            "$ref": "./examples/threatintelligence/CreateThreatIntelligence.json"
+          }
+        },
+        "tags": [
+          "ThreatIntelligence"
+        ],
+        "description": "Create a new threat intelligence indicator.",
+        "operationId": "ThreatIntelligenceIndicator_CreateIndicator",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../common/1.0/types.json#/parameters/SubscriptionId"
+          },
+          {
+            "$ref": "../../../common/1.0/types.json#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "../../../common/1.0/types.json#/parameters/OperationalInsightsResourceProvider"
+          },
+          {
+            "$ref": "../../../common/1.0/types.json#/parameters/WorkspaceName"
+          },
+          {
+            "$ref": "#/parameters/ThreatIntelligenceProperties"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ThreatIntelligenceInformation"
+            }
+          },
+          "201": {
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/ThreatIntelligenceInformation"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed to create indicators.",
+            "schema": {
+              "$ref": "../../../common/1.0/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{operationalInsightsResourceProvider}/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/threatIntelligence/main/indicators": {
```

Comment:
>threatIntelligence/main/indicators [](http://example.com/codeflow?start=184&length=34)

This is not a valid ARM resource model. You should model this with alternating type and instance segments. This will prevent ARG and Azure Policy from working with these resources.

Wiki links:
[https://armwiki.azurewebsites.net/introduction/concepts/resourceids.html](https://armwiki.azurewebsites.net/introduction/concepts/resourceids.html)
[https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc004-uri-is-not-following-arm-standard-guidelines](https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc004-uri-is-not-following-arm-standard-guidelines).

Reply:
@mentat9: According to suggestions of ARM Reviewer in our RP in preview version, we are treating {threatIntelligence} as  type and {main} as singleton resource
Previous PR: https://github.com/Azure/azure-rest-api-specs/pull/10463

Reply:
OK, that's fine, but you haven't implemented operations for the parent type (Microsoft.SecurityInsights/threatIntelligence). It's not possible for a client like ARG or Azure Policy to discover this singleton.

Reply:
Previously reviewed.